mentoring/mcq.py

```diff
@@ -0,0 +1,197 @@
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+
+from .questionnaire import QuestionnaireAbstractBlock
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+# Classes ###########################################################
+
+
+class MCQBlock(QuestionnaireAbstractBlock):
+    """
+    An XBlock used to ask multiple-choice questions
+    """
+    student_choice = String(
+        # {Last input submitted by the student
+        default="",
+        scope=Scope.user_state,
+    )
+
+    correct_choices = List(
+        display_name=_("Correct Choice[s]"),
+        help=_("Specify the value[s] that students may select for this question to be considered correct."),
+        scope=Scope.content,
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+    )
+    editable_fields = QuestionnaireAbstractBlock.editable_fields + ('correct_choices', )
+
+    def describe_choice_correctness(self, choice_value):
+        if choice_value in self.correct_choices:
+            if len(self.correct_choices) == 1:
+                # Translators: This is an adjective, describing a choice as correct
+                return self._(u"Correct")
+            return self._(u"Acceptable")
+        else:
+            if len(self.correct_choices) == 1:
+                return self._(u"Wrong")
+            return self._(u"Not Acceptable")
+
+    def submit(self, submission):
+        log.debug(u'Received MCQ submission: "%s"', submission)
+
+        correct = submission in self.correct_choices
+        tips_html = []
+        for tip in self.get_tips():
+            if submission in tip.values:
+                tips_html.append(tip.render('mentoring_view').content)
+
+        if tips_html:
+            formatted_tips = ResourceLoader(__name__).render_template('templates/html/tip_choice_group.html', {
+                'tips_html': tips_html,
+            })
+
+        self.student_choice = submission
+        result = {
+            'submission': submission,
+            'status': 'correct' if correct else 'incorrect',
+            'tips': formatted_tips if tips_html else None,
+            'weight': self.weight,
+            'score': 1 if correct else 0,
+        }
+        log.debug(u'MCQ submission result: %s', result)
+        return result
+
+    def author_edit_view(self, context):
+        """
+        The options for the 1-5 values of the Likert scale aren't child blocks but we want to
+        show them in the author edit view, for clarity.
+        """
+        fragment = Fragment(u"<p>{}</p>".format(self.question))
+        self.render_children(context, fragment, can_reorder=True, can_add=False)
```

Comment:
Do you have an overridden `render_children` somewhere that injects data into fragment, and uses `can_reorder` and `can_add`? I wasn't seeing it in this PR or the repo, but I don't know where to look.